p2p/client.go

```diff
@@ -394,17 +503,43 @@ func (c *Client) NewTxValidator() GossipValidator {
 	}
 }
 
-func (c *Client) gossipedBlockReceived(msg *GossipMessage) {
-	var gossipedBlock GossipedBlock
-	if err := gossipedBlock.UnmarshalBinary(msg.Data); err != nil {
+// blockSyncReceived is called on reception of new block via block-sync protocol
+func (c *Client) blockSyncReceived(block *P2PBlockEvent) {
+	err := c.localPubsubServer.PublishWithEvents(context.Background(), *block, map[string][]string{EventTypeKey: {EventNewBlockSyncBlock}})
+	if err != nil {
+		c.logger.Error("Publishing event.", "err", err)
+	}
+	// Received block is cached and  no longer needed to request using block-sync
+	if c.conf.BlockSyncEnabled {
+		c.addBlockReceived(block.Block.Header.Height)
+	}
+}
+
+// blockSyncReceived is called on reception of new block via gossip protocol
+func (c *Client) blockGossipReceived(ctx context.Context, block []byte) {
+	var gossipedBlock P2PBlockEvent
+	if err := gossipedBlock.UnmarshalBinary(block); err != nil {
 		c.logger.Error("Deserialize gossiped block", "error", err)
 	}
 	err := c.localPubsubServer.PublishWithEvents(context.Background(), gossipedBlock, map[string][]string{EventTypeKey: {EventNewGossipedBlock}})
 	if err != nil {
 		c.logger.Error("Publishing event.", "err", err)
 	}
+	if c.conf.BlockSyncEnabled {
```

Comment:
same question for here. I see in multiple places `c.conf.BlockSyncEnabled`. is there a way to make it only on one place? 

seems like it's a generic function, so I guess we need it and in that case I think cleaner just 
`if !c.conf.BlockSyncEnabled {
    return
 }`

Reply:
done